VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -346,11 +356,17 @@ protected void onException(Exception ex) {
                 nextPage = 1;
             }
         };
-        getThreadListTask.setProgressCallback(null);
+        // Initially we need to show the spinner at the center of the screen. After that the
+        // ListView will start showing a footer-based loading indicator.
+        if (nextPage > 1 || callback.isRefreshingSilently()) {
+            getThreadListTask.setProgressCallback(null);
+        } else {
+            getThreadListTask.setProgressDialog(loadingIndicator);
+        }
         getThreadListTask.execute();
     }
 
-    private void setNoResultView(@Nullable EmptyQueryResultsFor query) {
+    private void setScreenStateUponResult(@Nullable EmptyQueryResultsFor query) {
```

Comment:
The parameter is defined as `Nullable`, yet there is no null-check on it, which produces warning. I suppose you should add a `null` check on it.